mne/inverse_sparse/mxne_inverse.py

```diff
@@ -161,11 +162,104 @@ def _make_sparse_stc(X, active_set, forward, tmin, tstep,
 
 
 @verbose
+def _make_dipoles_sparse(X, active_set, forward, tmin, tstep, M, M_est,
+                         active_is_idx=False, verbose=None):
+
+    times = tmin + tstep * np.arange(X.shape[1])
+
+    if not active_is_idx:
+        active_idx = np.where(active_set)[0]
+    else:
+        active_idx = active_set
+
+    n_dip_per_pos = 1 if is_fixed_orient(forward) else 3
+    if n_dip_per_pos > 1:
+        active_idx = np.unique(active_idx // n_dip_per_pos)
+
+    gof = np.zeros(M_est.shape[1])
+    M_norm = np.sqrt(np.sum(M ** 2, axis=0))
+    gof[M_norm > 0.0] = (np.sqrt(np.sum(M_est ** 2., axis=0))[M_norm > 0.0] /
+                         M_norm[M_norm > 0.0] * 100.)
+
```

Comment:
I am bit concerned that GOF for each dipole is not meaningful with such multi dipole fit approach?

wdyt of having a GOF which is dipole specific ?

basically to store the variance each dipole can explain individually